contracts/Quest.sol

```diff
@@ -164,6 +176,17 @@ contract Quest is ReentrancyGuardUpgradeable, PausableUpgradeable, Ownable, IQue
         questFactoryContract.withdrawCallback(questId, protocolFeeRecipient, protocolPayout, address(owner()), ownerPayout);
     }
 
+    function claimReferralFees(address referrer) external onlyWithdrawAfterEnd {
+        if (referrerHasClaimed[referrer] == true) revert AlreadyWithdrawn();
+
+        uint256 referrerClaimAmount = referralClaimAmounts[referrer];
+        if (referrerClaimAmount > 0) {
```

Comment:
I would either do this check as a revert to prevent the tx from even getting pushed or just remove it altogether.